pkg/settlement/pseudosettle/pseudosettle.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package pseudosettle
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/p2p"
+	"github.com/ethersphere/bee/pkg/p2p/protobuf"
+	"github.com/ethersphere/bee/pkg/settlement"
+	pb "github.com/ethersphere/bee/pkg/settlement/pseudosettle/pb"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+const (
+	protocolName    = "pseudosettle"
+	protocolVersion = "1.0.0"
+	streamName      = "pseudosettle"
+)
+
+type Service struct {
+	streamer p2p.Streamer
+	logger   logging.Logger
+	observer settlement.PaymentObserver
+}
+
+type Options struct {
+	Streamer p2p.Streamer
+	Logger   logging.Logger
+}
+
+func New(o Options) *Service {
+	return &Service{
+		streamer: o.Streamer,
+		logger:   o.Logger,
+	}
+}
+
+func (s *Service) Protocol() p2p.ProtocolSpec {
+	return p2p.ProtocolSpec{
+		Name:    protocolName,
+		Version: protocolVersion,
+		StreamSpecs: []p2p.StreamSpec{
+			{
+				Name:    streamName,
+				Handler: s.handler,
+			},
+		},
+	}
+}
+
+func (s *Service) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream) (err error) {
+	r := protobuf.NewReader(stream)
+	defer func() {
+		if err != nil {
+			_ = stream.Reset()
+		} else {
+			_ = stream.FullClose()
+		}
+	}()
+	var req pb.Payment
+	if err := r.ReadMsg(&req); err != nil {
+		return fmt.Errorf("read request: %w peer %v", err, p.Address)
```

Comment:
Chained error in error message is usually at the end.

Reply:
fixed. I took this part of the code from `retrieval`. It should probably be fixed there as well for consistent log messages.

Reply:
If it is a small fix, I would suggest just to do it right away. If it is a bigger task, it would be good to create an issue (`cleanup`).